Change shear mixing to LMD94 formulation and add ri smoothing (#352)
E3SM-Project/Omega: components/omega/doc/devGuide/VerticalMixingCoeff.md

# Vertical Mixing Coefficients

Omega includes a `VertMix` class that provides functions that compute `VertDiff`, `VertVisc`, `GradRichNum`, and `GradRichNumSmoothed`, the

> vertical diffusivity and viscosity, the gradient Richardson number, a smoothed gradient Richardson number, where all are defined at the center of the cell and top of the layer.

Reviewer: Let's specify that we smooth the gradient richardson number with a 1-2-1 filter.

Reviewer: oops I see the specification just down a bit

Currently the values of `VertDiff` and `VertVisc` are calculated using the linear combination of three options: (1) a
constant background mixing value, (2) a convective instability mixing value, and (3) a Richardson

> number dependent shear mixing value from the [Large et al (1994)](https://agupubs.onlinelibrary.wiley.com/doi/epdf/10.1029/94RG01872) or LMD94 interior shear parameterization. These options are linearly additive. In the future, additional additive options will be implemented, such as the K Profile Parameterization [(KPP; Large et al., 1994)](https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/94rg01872). For both the convective and shear mixing values `BruntVaisalaFreq` is needed, which

Reviewer: this is a little pedantic, but let's change `shear mixing` -> `shear instability driven mixing` throughout.

> is calculated by the `EOS` class. `GradRichNum` is smoothed using a 1-2-1 filter to produce `GradRichNumSmoothed` which is used by the LMD94 shear instability driven mixing formulation.

Reviewer 2: ```suggestion
is calculated by the `EOS` class. `GradRichNum` is smoothed using a 1-2-1 (vertical) filter to produce `GradRichNumSmoothed` which is used by the LMD94 shear instability driven mixing formulation.
```

    TriggerBVF: 0.0
  Shear:
    Enable: true
    BaseShearValue: 0.005
    RiCrit: 0.7
    Exponent: 3.0

>     RiSmoothLoops: 3

Reviewer: again change to 2

3. Shear Mixing:
   - `EnableShearMix`: Flag to enable/disable shear mixing (Default: True)

>    - `BaseShearValue`: Base values of shear for the LMD94 interior shear mixing formulation (Default: 0.005)

Reviewer: ```suggestion
   - `BaseShearValue`: Base values of maximum viscosity and diffusivity for the LMD94 interior shear instability driven mixing formulation (Default: 0.005)
```

>    - `RiCrit`: Critical Richerson number for the LMB94 formulation (Default: 0.7)

Reviewer: ```suggestion
   - `RiCrit`: Critical Richardson number for the LMD94 formulation (Default: 0.7)
```

>    - `ShearExponent`: Exponent parameter number for the LMB94 formulation (Default: 3.0)

Reviewer: ```suggestion
   - `ShearExponent`: Exponent parameter number for the LMD94 formulation (Default: 3.0)
```

>    - `RiSmoothLoops`: Number of 1-2-1 filter passes to apply to the gradient Richardson number smoothing (Default: 3)

Reviewer: ```suggestion
   - `RiSmoothLoops`: Number of 1-2-1 filter passes to apply to the gradient Richardson number smoothing (Default: 2)
```
